Sources/Layout/MessagesCollectionViewFlowLayout.swift

```diff
@@ -206,44 +473,26 @@ extension MessagesCollectionViewFlowLayout {
     }
 }
 
-// MARK: - MessageCellLayoutContext
+// MARK: - Helpers
 
 extension MessagesCollectionViewFlowLayout {
 
-    internal func cellLayoutContext(for message: MessageType, at indexPath: IndexPath) -> MessageCellLayoutContext {
-        guard let cachedContext = layoutContextCache.object(forKey: message.messageId as NSString) else {
-            let newContext = newCellLayoutContext(for: message, at: indexPath)
+    internal func labelSize(for attributedText: NSAttributedString, considering maxWidth: CGFloat) -> CGSize {
 
-            if messagesLayoutDelegate.shouldCacheLayoutAttributes(for: message) {
-                layoutContextCache.setObject(newContext, forKey: message.messageId as NSString)
-            }
-            return newContext
-        }
-        return cachedContext
-    }
-
-    internal func newCellLayoutContext(for message: MessageType, at indexPath: IndexPath) -> MessageCellLayoutContext {
-        currentLayoutContext = MessageCellLayoutContext()
-        currentLayoutContext.avatarPosition = _avatarPosition(for: message, at: indexPath)
-        currentLayoutContext.avatarSize = _avatarSize(for: message, at: indexPath)
-        currentLayoutContext.messageContainerPadding = _messageContainerPadding(for: message, at: indexPath)
-        currentLayoutContext.messageLabelInsets = _messageLabelInsets(for: message, at: indexPath)
-        currentLayoutContext.messageContainerMaxWidth = _messageContainerMaxWidth(for: message, at: indexPath)
-        currentLayoutContext.messageContainerSize = _messageContainerSize(for: message, at: indexPath)
-        currentLayoutContext.topLabelAlignment = _cellTopLabelAlignment(for: message, at: indexPath)
-        currentLayoutContext.topLabelMaxWidth = _cellTopLabelMaxWidth(for: message, at: indexPath)
-        currentLayoutContext.topLabelSize = _cellTopLabelSize(for: message, at: indexPath)
-        currentLayoutContext.bottomLabelAlignment = _cellBottomLabelAlignment(for: message, at: indexPath)
-        currentLayoutContext.bottomLabelMaxWidth = _cellBottomLabelMaxWidth(for: message, at: indexPath)
-        currentLayoutContext.bottomLabelSize = _cellBottomLabelSize(for: message, at: indexPath)
-        currentLayoutContext.itemHeight = _cellContentHeight(for: message, at: indexPath)
-        return currentLayoutContext
-    }
-}
+        guard let cachedSize = attributedTextSizeCache.object(forKey: attributedText.hash as NSNumber)?.cgSizeValue else {
```

Comment:
This won't work yet. We may have seen the exact same text on another message with a different width